go/vt/sqlparser/sql.y

```diff
@@ -870,9 +870,13 @@ value_expression:
   {
     $$ = &BinaryExpr{Left: $1, Operator: JSONUnquoteExtractOp, Right: $3}
   }
-| value_expression COLLATE value_expression
+| value_expression COLLATE STRING
```

Comment:
channeling my inner @sougou, I would simplify these 2 statements into 1. You can either use one of the pre-existing types for the right-hand side and let mysql handle the validation, or change to something like:

```yacc
... snip ...
  value_expression COLLATE charset
  {
    $$ = &CollateExpr{Expr: $1, Charset: $3}
  }
charset
  ID
  {
    $$ = string($1)
  }
| STRING
  {
    $$ = string($2)
  }
```